### ISSUE TITLE
VideoCoCa: Video-Text Modeling with Zero-Shot Transfer from Contrastive Captioners

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#220]
# TL; DR

- Extend CoCa along the temporal dimension.

# Motivation

- Image-text foundation models achieve strong performance in diverse zero-shot / finetuned benchmarks. What about on videos?

# Formulation

<img width="917" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/a8173c3e-9ca5-45ae-8630-1834a61ee201">

### Architectural Design

- Attentional Poolers

> Flatten image patch tokens along the temporal dimension: (BxT)xNxd -> Bx(TxN)xd and apply attention pooling of CoCa.

- Factorized Encoder

> Learn additional transformer encoder on top of attention-pooled [CLS] token (BxTxd).

- Joint Space-time Encoder

> Gridify images across temporal dimension: (BxT)xNxd -> Bx(TxN)xd **before** feeding to the **image encoder** (with TxNxd dim positional encoding).

- Mean Pooling

> Process frames as (BxT)xNxd until attention pooling and then simply average-pool along the temporal dimension.

### Training Scheme

- FT (full finetuning)

- Frozen Encoder Decoder tuning (only attention poolers)

- Frozen + FT (first attention poolers, and then full FT)

- LiT (freeze image encoder)

# Evaluation

<img width="487" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/b286d5c4-e1c4-49aa-9c18-3caba2cc89e7">
<img width="491" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/7beaef67-0cd6-4af5-8546-def285a3283c">

- Attention pooler works the best (despite its simplicity).
- Early fusing causes domain deviation (from natural image to image grid, similar finding in #186 ).
- LiT performs well.
- As CoCa is already a very strong vision-language learner, maximally preserving its pretrained knowledge helps. Especially for tasks like video classification and video retrieval (where a single [CLS] token matters), CoCa itself can be a very strong baseline.

<img width="1006" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/5512da97-c7b0-496a-b7c7-bf52077a4c97">

- Simply rearranging the attention poolers (VideoCoCa w.o. videos) improves performances non-trivially.

<img width="1016" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/d9ea33e5-b096-4e6e-ba3b-9a14aabe8a55">
<img width="1004" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/447348ff-5d28-415d-9131-d510db8498e2">
<img width="502" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/b27acf3a-58f6-4550-8619-720bf7a0904d">

# Inspiration

- Attention pooling can be a really strong and versatile design choice to model different problem scenarios.
- Hastily gridifying images usually hurts previous knowledge.
